src/main/java/seedu/loyaltylift/logic/commands/EditOrderCommand.java

```diff
@@ -0,0 +1,186 @@
+            + "by the index number used in the displayed order list. "
+            + "Existing values will be overwritten by the input values. "
+            + "Note, if you'd like to edit the status, use the advo and revo commands. \n"
+            + "Parameters: ORDER_INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_QUANTITY + "QUANTITY] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Signature Fudgy Brownies "
+            + PREFIX_QUANTITY + "4 "
+            + PREFIX_ADDRESS + "5 Bishan Road";
+
+    public static final String MESSAGE_EDIT_ORDER_SUCCESS = "Edited Order: \n%1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final Index index;
+    private final EditOrderDescriptor editOrderDescriptor;
+
+    /**
+     * @param index of the order in the filtered order list to edit
+     * @param editOrderDescriptor details to edit the order with
+     */
+    public EditOrderCommand(Index index, EditOrderDescriptor editOrderDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editOrderDescriptor);
+
+        this.index = index;
+        this.editOrderDescriptor = new EditOrderDescriptor(editOrderDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Order> lastShownList = model.getFilteredOrderList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+        }
+
+        Order orderToEdit = lastShownList.get(index.getZeroBased());
+        Order editedOrder = createEditedOrder(orderToEdit, editOrderDescriptor);
+
+        model.setOrder(orderToEdit, editedOrder);
+        model.updateFilteredOrderList(PREDICATE_SHOW_ALL_ORDERS);
+        return new CommandResult(String.format(MESSAGE_EDIT_ORDER_SUCCESS, editedOrder));
+    }
+
+    /**
+     * Creates and returns a {@code Order} with the details of {@code orderToEdit}
+     * edited with {@code editOrderDescriptor}.
+     */
+    private static Order createEditedOrder(
+            Order orderToEdit, EditOrderDescriptor editOrderDescriptor) {
+        assert orderToEdit != null;
+
+        Customer currentCustomer = orderToEdit.getCustomer();
+        Name updatedName = editOrderDescriptor.getName().orElse(orderToEdit.getName());
+        Status currentStatus = orderToEdit.getStatus();
+        Quantity updatedQuantity = editOrderDescriptor.getQuantity().orElse(orderToEdit.getQuantity());
+        Address updatedAddress = editOrderDescriptor.getAddress().orElse(orderToEdit.getAddress());
+
+        return new Order(currentCustomer, updatedName, updatedQuantity, currentStatus, updatedAddress);
```

Comment:
I believe this resets the Order's `CreatedDate` instead of using the original Order's `CreatedDate`